src/style/fog_helpers.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import {vec3} from 'gl-matrix';
+import MercatorCoordinate from '../geo/mercator_coordinate.js';
+import {smoothstep} from '../util/util.js';
+import type LngLat from '../geo/lng_lat.js';
+import type {UnwrappedTileID} from '../source/tile_id.js';
+import type Transform from '../geo/transform.js';
+
+export const FOG_PITCH_START = 55;
+export const FOG_PITCH_END = 65;
+export const FOG_SYMBOL_CLIPPING_THRESHOLD = 0.9;
+
+export type FogState = {
+    range: [number, number],
+    strength: number
+};
+
+// As defined in _prelude_fog.fragment.glsl#fog_opacity
+export function getFogOpacity(state: FogState, depthPx: number, pitch: number): number {
+    const fogOpacity = smoothstep(FOG_PITCH_START, FOG_PITCH_END, pitch);
+    const [start, end] = state.range;
+    const fogStrength = state.strength;
+
+    // The fog is not physically accurate, so we seek an expression which satisfies a
+    // couple basic constraints:
+    //   - opacity should be 0 at the near limit
+    //   - opacity should be 1 at the far limit
+    //   - the onset should have smooth derivatives to avoid a sharp band
+    // To this end, we use an (1 - e^x)^n, where n is set to 3 to ensure the
+    // function is C2 continuous at the onset. The fog is about 99% opaque at
+    // the far limit, so we simply scale it and clip to achieve 100% opacity.
+    // https://www.desmos.com/calculator/3taufutxid
+    // The output of this function should match src/shaders/_prelude_fog.fragment.glsl
+    const decay = 6;
+    const t = (depthPx - start) / (end - start);
```

Comment:
Things may be in flux, but at the moment, this appears to be a comparison between depth in pixels and start/end in window-height-fractions. I think the two options are either that depth is in units of window-height-fractions (in which case this should be `depth`, not `depthPx`), or range should have pixels multiplied into it (which I don't believe it currently does).